src/test/java/com/google/sps/servlets/MutualFriendsServletTest.java

```diff
@@ -0,0 +1,166 @@
+    addTestUserEntityToDatastore(TEST_USER_1_ID, /* friendsList= */ new String[]{TEST_USER_3_ID, TEST_USER_4_ID});
+    addTestUserEntityToDatastore(TEST_USER_2_ID, /* friendsList= */ new String[]{});
+
+    String jsonOutput = execute(TEST_USER_1_ID, TEST_USER_2_ID);
+    List<String> mutualFriends = Arrays.asList(new Gson().fromJson(jsonOutput, String[].class));
+
+    assertThat(mutualFriends).isEmpty();
+  }
+
+  @Test
+  public void noMutualFriendsFilledLists() throws Exception {
+    addTestUserEntityToDatastore(TEST_USER_1_ID, /* friendsList= */ new String[]{TEST_USER_3_ID, TEST_USER_4_ID});
+    addTestUserEntityToDatastore(TEST_USER_2_ID, /* friendsList= */ new String[]{TEST_USER_5_ID});
+
+    String jsonOutput = execute(TEST_USER_1_ID, TEST_USER_2_ID);
+    List<String> mutualFriends = Arrays.asList(new Gson().fromJson(jsonOutput, String[].class));
+
+    assertThat(mutualFriends).isEmpty();
+  }
+
+  @Test
+  public void singleMutualFriend() throws Exception {
+    addTestUserEntityToDatastore(TEST_USER_1_ID, /* friendsList= */ new String[]{TEST_USER_3_ID});
+    addTestUserEntityToDatastore(TEST_USER_2_ID, /* friendsList= */ new String[]{TEST_USER_3_ID});
+
+    String jsonOutput = execute(TEST_USER_1_ID, TEST_USER_2_ID);
+    List<String> mutualFriends = Arrays.asList(new Gson().fromJson(jsonOutput, String[].class));
+
+    assertThat(mutualFriends).containsExactly(TEST_USER_3_ID);
+  }
+
+  @Test
+  public void multipleMutualFriends() throws Exception {
+    addTestUserEntityToDatastore(TEST_USER_1_ID, /* friendsList= */ new String[]{TEST_USER_3_ID, TEST_USER_4_ID, TEST_USER_5_ID});
+    addTestUserEntityToDatastore(TEST_USER_2_ID, /* friendsList= */ new String[]{TEST_USER_3_ID, TEST_USER_4_ID});
+
+    String jsonOutput = execute(TEST_USER_1_ID, TEST_USER_2_ID);
+    List<String> mutualFriends = Arrays.asList(new Gson().fromJson(jsonOutput, String[].class));
+
+    assertThat(mutualFriends).containsExactly(TEST_USER_3_ID, TEST_USER_4_ID);
+  }
+
+  /**
+   * Method that calls on the MutualFriendsServlet and returns the set of mutual friends
+   * between two users.
+   *
+   * @param userID1 One of the users in the match whose mutual friends are being found
+   * @param userID2 The other user in the match whose mutual friends are being found
+   * @return The response from the get request to MutualFriendsServlet as a string
+   */
+  private String execute(String userID1, String userID2) throws IOException {
+    when(mockRequest.getParameter(MutualFriendsServlet.USER_ID_1_REQUEST_URL_PARAM)).thenReturn(userID1);
+    when(mockRequest.getParameter(MutualFriendsServlet.USER_ID_2_REQUEST_URL_PARAM)).thenReturn(userID2);
+    responseWriter = new StringWriter();
+    writer = new PrintWriter(responseWriter, true);
+    when(mockResponse.getWriter()).thenReturn(writer);
+
+    servletUnderTest.doGet(mockRequest, mockResponse);
+    
+    return responseWriter.toString();
+  }
```

Comment:
supernit: I would move this below `addTestuserEntityToDatastore` since it is invoked *after* that method

Reply:
Got it!